File: backend/src/main/java/com/vetflow/api/domain/port/PatientRepository.java
package com.vetflow.api.domain.port;

import java.util.List;
import java.util.Optional;

import com.vetflow.api.domain.model.Patient;

public interface PatientRepository {
    Patient save(Patient patient);
    Optional<Patient> findById(Long id);
    List<Patient> findAll();
    List<Patient> findByOwnerId(Long ownerId);
    List<Patient> findByActive(boolean active);
    List<Patient> findByOwnerIdAndActive(Long ownerId, boolean active);
    void deleteById(Long id);
}

File: frontend/src/domain/ports/repositories/PatientRepository.ts
import type { Patient } from '@domain/models/Patient'

export interface PatientFilter {
  ownerId?: number
  species?: string
  isActive?: boolean
  searchTerm?: string // for name, microchip
}

export interface PatientRepository {
  findById(id: number): Promise<Patient | null>
  findAll(filter?: PatientFilter): Promise<Patient[]>
  findByOwnerId(ownerId: number): Promise<Patient[]>
  findByMicrochipNumber(microchipNumber: string): Promise<Patient | null>
  save(patient: Partial<Patient>): Promise<Patient>
  update(id: number, patient: Partial<Patient>): Promise<Patient>
  delete(id: number): Promise<void>
  deactivate(id: number): Promise<void>
}